### PR TITLE
Adding documentation to functions.

### PR DESCRIPTION
**Pull request by ShuvamDasWork**
The branch doc_streamline has added comments in the form of docstrings to all functions currently being used by the class MFF. The config file has also been changed in order to bring the theme of the html documentation closer to sktime's existing theme.

### CODE REVIEW

File: mff/mff.py [hunk 1]
```diff
@@ -370,25 +553,45 @@ def AddIslandsToConstraints(C:pd.DataFrame,
...
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Dataframe containing known values of all variables and nan for 
+        unknown values.
+    row : str
+        Row index of cell to be forecasted.
+    col : str
+        Column index of cell to be forecasted. 
+    forecaster : pipeline (?)
```
Comment (sakaiando) [inline mff/mff.py:569]: @ShuvamDasWork  What is the question about?
@SultanOrazbayevWork  If we allow users to specify single forecaster, pipeline, gridsearch, etc, should we restrict "forecaster" to be "BaseForecaster" or is there a better to restrict the object?

File: mff/mff.py [hunk 2]
```diff
@@ -276,6 +395,7 @@ def find_permissible_wildcard(constraints_with_wildcard):
 
 
     def find_strings_to_replace_wildcard(constraint,var_list,wildcard):
+        """"""
```
[the remark this answers is not in the document]
Comment (SultanOrazbayevWork) [reply mff/mff.py:398]: This is opening """ followed by nothing and the closing """. (usually used as a placeholder, but ends up as empty documentation)

File: mff/mff.py (hunk 1 above)
Comment (SultanOrazbayevWork) [reply mff/mff.py:569]: `BaseForecaster` is appropriate. Pipeline, GridSearch, etc. need to comply with BaseForecaster.

File: mff/mff.py [hunk 3]
```diff
@@ -776,29 +1111,61 @@ def GenSmoothingMatrix(W,lamstar):
...
-        DESCRIPTION.
-    Phi : TYPE
-        DESCRIPTION.
-    C : TYPE
-        DESCRIPTION.
-    d : TYPE
-        DESCRIPTION.
-    C_ineq : TYPE, optional
-        DESCRIPTION. The default is None.
+    y1 : Series
```
Comment (SultanOrazbayevWork) [inline mff/mff.py:1119]: Is this a Series or a DataFrame? (I'm not sure)
if this is a Series, let's use `pd.Series` for consistency across the docs.

File: mff/mff.py [hunk 4]
```diff
@@ -89,6 +91,47 @@ def DefaultForecaster()->BaseForecaster:
...
+     
+    ineq_constraints_with_wildcard : str, optional(default: None)
+        Inequality constraints, comparable to ``constraints_with_wildcard``.
+        Constraints may include wildcard, in which case constraints will be
+        applied across all horizons, or may be defined for specified time 
+        periods.
+       
+    parrallelize : boolean
+        Indicate whether paralellization should be employed for generating 
+        first step forecasts
```
Comment (SultanOrazbayevWork) [inline mff/mff.py:125]:
```suggestion
    parallelize : boolean
        Indicate whether parallelization should be employed for generating the
        first step forecasts. Default value is `True`.
```

**Review by SultanOrazbayevWork — CHANGES_REQUESTED** [review]
Overall, look's good, but let's discuss a few points next time:
- the target/intended structure for the module;
- consistency in docs (mostly type annotations).

File: docs/source/mff.rst
[the remark this answers is not in the document]
Comment (ShuvamDasWork) [reply docs/source/mff.rst:1]: Agreed, that is confusing. Currently I am using `mff_documentation.rst` to build the documentation, so this `mff.rst` file is not being used. We could delete this file.

**Review by sakaiando — APPROVED** [review]

File: mff/mff.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShuvamDasWork) [reply mff/mff.py]: I have updated the documentation for this part, so this is no longer a problem.

File: mff/mff.py [hunk 5]
```diff
@@ -409,16 +631,47 @@ def FillAnEmptyCell(df,row,col,forecaster):
...
+    df: pd.DataFrame
+        Dataframe containing known values of all variables and nan for 
+        unknown values.
+    
+    forecaster : BaseForecaster
+        sktime BaseForecaster descendant
+
+    parrallelize : boolean
+        Indicate whether paralellization should be employed for generating 
+        first step forecasts
```
Comment (SultanOrazbayevWork) [inline mff/mff.py:649]:
```suggestion
    parallelize : boolean
        Indicate whether parallelization should be employed for generating the
        first step forecasts. Default value is `True`. 
```

**Review by SultanOrazbayevWork — CHANGES_REQUESTED** [review]
A small consistency fix is needed, re parallelize argument doc.

**Pull request merged**